include/SupportedUserTypes.h

```diff
@@ -285,31 +285,23 @@ namespace ChimeraTK {
 
   template<typename NUMERIC>
   NUMERIC detail::userTypeToNumeric_impl<std::string, NUMERIC>::impl(const std::string& value) {
-    bool isHexidecimal = value.length() > 1 and (value[1] == 'x' or value[1] == 'X'); // OxABCD form hex
-
-    if constexpr(std::is_same<NUMERIC, Boolean>::value) {
-      // special treatment for Boolean
-      std::stringstream ss;
-      Boolean converted;
-      if(isHexidecimal) {
-        ss << std::hex << value;
-      }
-      else {
-        ss.str(value);
-      }
-      ss >> converted;
-      return converted;
+
+    //If value starts with "0x" indicating hexidecimal
+    if(value.length() >= 2 and (value[0] == '0') and ((value[1] == 'x') or (value[1] == 'X'))) {
+        unsigned long long temp;
+        try {
+            temp = std::stoull(value, nullptr, 16);
+        }
+        catch(...) { // ignore any parsing errors and just return 0 in that case
+            temp = 0;
+        }
+        return detail::userTypeToNumeric_impl<unsigned long long, NUMERIC>::impl(temp);
```

Comment:
What happened to the previous implementation with the std::hex in the stringstream? That one was much better because it does not use an intermediate data type.

We have to discuss again what the wanted behaviour is (see review comment of the test file).